moveit_ros/moveit_servo/src/pose_tracking.cpp

```diff
@@ -238,6 +238,9 @@ void PoseTracking::targetPoseCallback(const geometry_msgs::PoseStampedConstPtr&
       geometry_msgs::TransformStamped target_to_planning_frame = transform_buffer_.lookupTransform(
           planning_frame_, target_pose_.header.frame_id, ros::Time(0), ros::Duration(0.1));
       tf2::doTransform(target_pose_, target_pose_, target_to_planning_frame);
+
+      // Prevent doTransform from copying a stamp of 0, which will cause the haveRecentTargetPose check to fail servo motions
+      target_pose_.header.stamp = msg->header.stamp;
```

Comment:
Since the target pose and planning frame are guaranteed to be different frames and transform between the frames is the latest possible (since `ros::Time(0)` is used), there are two cases here.

Either the transform is fixed or the transform is one between two moving frames. However, in both cases, the latest transform is used and the target pose is transformed into the planning frame. So, after the transform, isn't the situation equivalent to us having a target pose in the planning frame with a timestamp of `ros::Time::now()`?

Suppose we had a target pose in `odom` at timestamp 10, and we received it at timestamp 100 and transformed it to `base_link` with the latest transform, wouldn't the correct timestamp for the pose be 100 instead of 10? Probably doesn't make a world of difference... perhaps it doesn't make any difference, but it feels like it is more correct nonetheless.

Reply:
That is a good point. I don't think it will have impact here, but is the pattern people should follow elsewhere where it could make an impact.